lib/protocol/bep_extensions.go

```diff
@@ -212,6 +213,26 @@ func (f FileInfo) isEquivalent(other FileInfo, ignorePerms bool, ignoreBlocks bo
 	return false
 }
 
+func FromFSFileInfo(fi fs.FileInfo, name string, filesystem fs.Filesystem) FileInfo {
```

Comment:
I haven't reviewed anything yet, just happened to scroll to the bottom and saw this - i don't think we want to add a dependency on `fs` from `protocol`, so probably better to have this helper where it's used... or maybe scanner?

Reply:
Scanner probably makes sense - I actually wanted to use it there but didn't for diff size reasons :)